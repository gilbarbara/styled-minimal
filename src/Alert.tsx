import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import {
  baseStyles,
  colorStyles,
  lineHeightStyles,
  sx,
  system,
  variantStyles,
} from './utils/system';

import { AlertProps } from './types';

export const StyledAlert = styled('div')<AlertProps>(
  props => {
    const { borderRadius, maxWidth, padding } = getTheme(props, 'alert');

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${px(borderRadius)};
      font-size: inherit;
      line-height: ${lineHeightStyles};
      max-width: ${px(maxWidth)};
      padding: ${padding};
      width: 100%;

      a {
        ${colorStyles};
      }
    `;
  },
  sx,
  system,
);

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => (
  <StyledAlert ref={ref} {...props} />
));

Alert.defaultProps = {
  dark: false,
  invert: false,
  role: 'alert',
  variant: 'primary',
};
