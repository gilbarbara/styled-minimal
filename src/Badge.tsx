import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system, variantStyles } from './utils/system';

import { BadgeProps } from './types';

export const StyledBadge = styled('sup')<BadgeProps>(
  props => {
    const { dot } = props;
    const { borderRadius, fontSize, fontWeight, lineHeight, padding, size, sizeDot } = getTheme(
      props,
      'badge',
    );

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${dot ? '100%' : px(borderRadius)};
      display: inline-flex;
      font-size: ${dot ? 0 : px(fontSize)};
      font-weight: ${fontWeight};
      justify-content: center;
      line-height: ${dot ? 0 : px(lineHeight)};
      height: ${px(dot ? sizeDot : size)};
      padding: ${dot ? 0 : `0 ${px(padding)}`};
      min-width: ${px(dot ? sizeDot : size)};
    `;
  },
  sx,
  system,
);

export const Badge = React.forwardRef<HTMLSpanElement, BadgeProps>((props, ref) => (
  <StyledBadge ref={ref} {...props} />
));

Badge.defaultProps = {
  dark: false,
  dot: false,
  invert: false,
  variant: 'primary',
};
