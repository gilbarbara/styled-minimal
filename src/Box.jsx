import PropTypes from 'prop-types';
import styled from 'styled-components';

import {
  alignContent,
  alignItems,
  alignSelf,
  backgroundImage,
  borders,
  borderRadius,
  bottom,
  color,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontFamily,
  fontSize,
  fontStyle,
  fontWeight,
  height,
  justifyContent,
  left,
  lineHeight,
  maxHeight,
  maxWidth,
  minHeight,
  minWidth,
  order,
  position,
  ratio,
  right,
  space,
  textAlign,
  top,
  width,
  zIndex,
} from 'styled-system';

import { textTransform } from './utils/system';

export const Box = styled.div(
  {
    boxSizing: 'border-box',
  },
  alignContent,
  alignItems,
  alignSelf,
  backgroundImage,
  borders,
  borderRadius,
  bottom,
  color,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontFamily,
  fontSize,
  fontStyle,
  fontWeight,
  height,
  justifyContent,
  left,
  lineHeight,
  maxHeight,
  maxWidth,
  minHeight,
  minWidth,
  order,
  position,
  ratio,
  right,
  space,
  textAlign,
  textTransform,
  top,
  width,
  zIndex
);

Box.displayName = 'Box';

Box.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node,
  textTransform: PropTypes.string,
  ...alignContent.propTypes,
  ...alignItems.propTypes,
  ...alignSelf.propTypes,
  ...backgroundImage.propTypes,
  ...borders.propTypes,
  ...borderRadius.propTypes,
  ...bottom.propTypes,
  ...color.propTypes,
  ...display.propTypes,
  ...flex.propTypes,
  ...flexBasis.propTypes,
  ...flexDirection.propTypes,
  ...flexWrap.propTypes,
  ...fontFamily.propTypes,
  ...fontSize.propTypes,
  ...fontWeight.propTypes,
  ...height.propTypes,
  ...justifyContent.propTypes,
  ...left.propTypes,
  ...maxHeight.propTypes,
  ...maxWidth.propTypes,
  ...minHeight.propTypes,
  ...minWidth.propTypes,
  ...order.propTypes,
  ...position.propTypes,
  ...right.propTypes,
  ...ratio.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
  ...top.propTypes,
  ...width.propTypes,
  ...zIndex.propTypes,
};

export default Box;
