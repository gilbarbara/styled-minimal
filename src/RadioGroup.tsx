import * as React from 'react';

import { FormGroup } from './FormGroup';
import { Radio } from './Input';

import { NumberOrString, RadioGroupProps } from './types';

export class RadioGroup extends React.PureComponent<
  RadioGroupProps,
  { selectedValue: NumberOrString }
> {
  constructor(props: RadioGroupProps) {
    super(props);

    this.state = {
      selectedValue: props.value || props.defaultValue!,
    };
  }

  static defaultProps = {
    disabled: false,
    size: 'md',
    variant: 'primary',
  };

  componentDidUpdate(prevProps: RadioGroupProps) {
    const { value } = this.props;

    if (value && prevProps.value !== value) {
      this.update(value);
    }
  }

  update = (selectedValue: NumberOrString) => {
    this.setState({ selectedValue });
  };

  handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
    const { value, disabled, onChange } = this.props;
    const numericValue = Number(target.value);
    const currentValue = !Number.isNaN(numericValue) ? numericValue : target.value;

    if (onChange) {
      onChange(currentValue);
    }

    if (!value && !disabled) {
      this.update(currentValue);
    }
  };

  render() {
    const { selectedValue } = this.state;
    const {
      dark,
      defaultValue,
      disabled,
      inline,
      invert,
      name,
      onChange,
      options,
      size,
      value,
      variant,
      ...rest
    } = this.props;

    if (!options || !options.length) {
      return null;
    }

    return (
      <FormGroup {...rest} role="radiogroup">
        {options.map(d => (
          <Radio
            key={d.value}
            checked={d.value === selectedValue}
            dark={dark}
            disabled={disabled || d.disabled}
            inline={inline}
            invert={invert}
            label={d.label || d.value}
            name={name}
            onChange={this.handleChange}
            size={size}
            value={d.value}
            variant={d.variant || variant}
          />
        ))}
      </FormGroup>
    );
  }
}
