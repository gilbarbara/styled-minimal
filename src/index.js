export { default as Alert } from './Alert';
export { default as Badge } from './Badge';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Code } from './Code';
export { default as Container } from './Container';
export { default as Embed } from './Embed';
export { default as Flex } from './Flex';
export { default as Fieldset } from './Fieldset';
export { default as Form } from './Form';
export { default as FormGroup } from './FormGroup';
export { default as Group } from './Group';
export { default as Heading } from './Heading';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Legend } from './Legend';
export { default as Link } from './Link';
export { default as List } from './List';
export { default as Paragraph } from './Paragraph';
export { default as Screen } from './Screen';
export { default as Select } from './Select';
export { default as Switch } from './Switch';
export { default as Table } from './Table';
export { default as Text } from './Text';
export { default as Textarea } from './Textarea';
