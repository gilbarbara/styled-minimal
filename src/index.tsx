export { Alert } from './Alert';
export { Badge } from './Badge';
export { Box } from './Box';
export { Button } from './Button';
export { ButtonGroup } from './ButtonGroup';
export { Code } from './Code';
export { CodeBlock } from './CodeBlock';
export { Container } from './Container';
export { Embed } from './Embed';
export { Flex } from './Flex';
export { Fieldset } from './Fieldset';
export { Form } from './Form';
export { FormGroup } from './FormGroup';
export { Grid } from './Grid';
export { Group } from './Group';
export { Heading } from './Heading';
export { Image } from './Image';
export { Checkbox, Input, Radio } from './Input';
export { Label } from './Label';
export { Legend } from './Legend';
export { Link } from './Link';
export { DL, OL, UL } from './List';
export { Loader } from './Loader';
export { Paragraph } from './Paragraph';
export { RadioGroup } from './RadioGroup';
export { Select } from './Select';
export { Switch } from './Switch';
export { Table } from './Table';
export { Text } from './Text';
export { Textarea } from './Textarea';

export { default as system } from '@styled-system/css';
export { mergeTheme as getTheme, px, responsive } from './utils/helpers';
