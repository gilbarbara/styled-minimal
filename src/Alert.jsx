import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { size } = props;
  const { borderRadius, maxWidth, padding } = themeGet(props, 'alert');

  return css`
    ${baseStyles.variant};
    border-radius: ${px(borderRadius)};
    font-size: ${baseStyles.fontSize};
    line-height: ${baseStyles.lineHeight};
    max-width: ${px(maxWidth)};
    padding: ${px(padding[size][0])} ${px(padding[size][1])};
    width: 100%;

    a {
      ${baseStyles.color};
    }
  `;
};

const Alert = styled(Box).attrs({
  role: 'alert',
})`
  ${styles};
`;

Alert.displayName = 'Alert';

Alert.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  /** button size */
  size: sizesAllPropTypes,
  /** button variant */
  variant: variantPropTypes,
  ...Box.propTypes,
};

Alert.defaultProps = {
  outline: false,
  size: 'md',
  variant: 'primary',
};

export default Alert;
