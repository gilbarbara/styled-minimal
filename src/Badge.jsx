import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const Badge = styled(Box)`
  ${(props: Object): string => {
    const {
      borderRadius,
      fontSize,
      fontWeight,
      padding,
    } = themeGet(props, 'badge');
    const fontSizeProp = themeGet(props, 'componentSizes', { key: 'size' });

    return css`
      ${baseStyles.variant};
      border-radius: ${px(borderRadius)};
      display: inline-flex;
      font-size: ${fontSize};
      ${fontSizeProp ? `font-size: ${fontSizeProp}` : ''};
      font-weight: ${fontWeight};
      line-height: 1;
      padding: ${px(padding[0])} ${px(padding[1])};
      vertical-align: baseline;
    `;
  }};
`;

Badge.displayName = 'Badge';

Badge.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Badge.defaultProps = {
  as: 'span',
};

export default Badge;
