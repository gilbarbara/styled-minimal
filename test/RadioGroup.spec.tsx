import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { RadioGroup } from '../src/RadioGroup';

const mockOnChange = jest.fn();

describe('RadioGroup', () => {
  it('should render properly', () => {
    const { container } = render(
      <RadioGroup
        defaultValue={1}
        label="RadioGroup"
        name="test"
        options={[{ label: 'Option 1', value: 1 }, { value: 2 }]}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should work in uncontrolled mode', () => {
    const { container, getByLabelText } = render(
      <RadioGroup
        defaultValue="one"
        label="RadioGroup"
        name="test"
        onChange={mockOnChange}
        options={[
          { label: 'Option 1', value: 'one' },
          { label: 'Option 2', value: 'two' },
        ]}
      />,
    );

    const labels = container.querySelectorAll('label:not(:first-of-type)');
    const option1 = getByLabelText('Option 1') as HTMLInputElement;
    const option2 = getByLabelText('Option 2') as HTMLInputElement;

    expect(option1.checked).toBe(true);
    expect(option2.checked).toBe(false);

    fireEvent.click(labels[1]);

    expect(option1.checked).toBe(false);
    expect(option2.checked).toBe(true);

    expect(mockOnChange).toHaveBeenLastCalledWith('two');
  });

  it('should work in controlled mode', () => {
    const { container, getByLabelText, rerender } = render(
      <RadioGroup
        label="RadioGroup"
        name="test"
        options={[
          { label: 'Option 1', value: 1 },
          { label: 'Option 2', value: 2 },
        ]}
        value={1}
      />,
    );

    const labels = container.querySelectorAll('label:not(:first-of-type)');
    const option1 = getByLabelText('Option 1') as HTMLInputElement;
    const option2 = getByLabelText('Option 2') as HTMLInputElement;

    expect(option1.checked).toBe(true);
    expect(option2.checked).toBe(false);

    fireEvent.click(labels[1]);

    expect(option1.checked).toBe(true);
    expect(option2.checked).toBe(false);

    rerender(
      <RadioGroup
        label="RadioGroup"
        name="test"
        options={[
          { label: 'Option 1', value: 1 },
          { label: 'Option 2', value: 2 },
        ]}
        value={2}
      />,
    );

    expect(option1.checked).toBe(false);
    expect(option2.checked).toBe(true);
  });

  it('should not render if options is empty', () => {
    const { container } = render(
      <RadioGroup label="RadioGroup" name="test" options={[]} value={1} />,
    );

    expect(container.firstChild).toBeNull();
  });
});
