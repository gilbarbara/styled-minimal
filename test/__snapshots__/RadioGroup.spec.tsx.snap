// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should render properly 1`] = `
.c1 {
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2E2E2E;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: inherit;
  line-height: 1.4;
  margin-bottom: 8px;
  white-space: nowrap;
}

.c0 {
  box-sizing: border-box;
  margin-bottom: 16px;
  text-align: left;
}

.c0 label + label {
  margin-top: 6px;
}

.c0 .c8 input,
.c0 .c8 label,
.c0 .c8 legend {
  margin-bottom: 0;
  margin-right: 8px;
}

.c0 .c8 small {
  margin-top: 0;
}

.c4 {
  left: 0;
  opacity: 0;
  position: absolute;
  top: 0;
  z-index: -1;
}

.c7 {
  font-size: 16px;
}

.c6 {
  box-sizing: border-box;
  border-radius: 50%;
  background-color: #2E2E2E;
  border: 1px solid #2E2E2E;
  color: #fff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 16px;
  margin-right: 6px;
  position: relative;
  -webkit-transition: background-color 0.2s,border-color 0.2s;
  transition: background-color 0.2s,border-color 0.2s;
  width: 16px;
}

.c6:after {
  content: '';
  display: none;
  position: absolute;
  left: 50%;
  top: 50%;
  background-color: #fff;
  border-radius: 50%;
  height: 6px;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 6px;
}

input:checked ~ .c6:after {
  display: inline-block;
}

input:disabled ~ .c6 {
  opacity: 0.6;
}

input:not(:checked) ~ .c6 {
  background-color: #f7f8f9;
  border-color: #ccc;
}

input:not(:checked):disabled ~ .c6 {
  background-color: #ccc;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
  position: relative;
}

.c2:focus .c3:not(:checked) ~ .c5,
.c2:hover .c3:not(:checked) ~ .c5 {
  background-color: #ccc;
}

.c2:focus .c3:not(:disabled):checked ~ .c5,
.c2:hover .c3:not(:disabled):checked ~ .c5 {
  background-color: #949494;
  border-color: #949494;
}

<div
  class="c0"
  role="radiogroup"
>
  <label
    class="c1"
  >
    RadioGroup
  </label>
  <label
    class="c2"
  >
    <input
      checked=""
      class="c3 c4"
      name="test"
      type="radio"
      value="1"
    />
    <span
      aria-checked="true"
      class="c5 c6"
      role="radio"
    />
    <span
      class="c7"
    >
      Option 1
    </span>
  </label>
  <label
    class="c2"
  >
    <input
      class="c3 c4"
      name="test"
      type="radio"
      value="2"
    />
    <span
      aria-checked="false"
      class="c5 c6"
      role="radio"
    />
    <span
      class="c7"
    >
      2
    </span>
  </label>
</div>
`;
